addons/l10n_cl/models/account_move.py

```diff
@@ -109,11 +114,7 @@ def _get_starting_sequence(self):
     def _get_last_sequence_domain(self, relaxed=False):
         where_string, param = super(AccountMove, self)._get_last_sequence_domain(relaxed)
         if self.company_id.country_id == self.env.ref('base.cl') and self.l10n_latam_use_documents:
-            journals = self.journal_id.l10n_cl_sequence_ids.filtered(lambda s: s.l10n_latam_document_type_id == self.l10n_latam_document_type_id).l10n_cl_journal_ids.ids
-            if len(journals) > 1:
-                where_string.replace("journal_id = %(journal_ids)s", "journal_id in %(journal_ids)s")
-                param['journal_ids'] = journals
-            where_string += " AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s "
+            where_string = "WHERE l10n_latam_document_type_id = %(l10n_latam_document_type_id)s AND name != '/'"
```

Comment:
Not very inheritance friendly and you really need to take into account company_id

```suggestion
            where_string.replace("journal_id = %(journal_id)s", "")
            where_string += " AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s AND  company_id = %(company_id)"
            param['company_id'] = self.company_id.id or False
```

Reply:
Thanks @jco-odoo! I had problems with the replace method. It didn't work. I've added an equal ->
where_string = where_string.replace....

Otherwise, I can't add the company_id parameter, something goes wrong and I've not found the error yet.
Here the traceback:

```
odoo13_1  | 2020-09-29 01:20:10,393 1 ERROR sass13_v2 odoo.sql_db: bad query:
odoo13_1  |             UPDATE account_move SET write_date = write_date WHERE id = (
odoo13_1  |                 SELECT id FROM account_move
odoo13_1  |                 WHERE  name != '/' AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s AND company_id = %(company_id)
odoo13_1  |                 AND sequence_prefix = (SELECT sequence_prefix FROM account_move WHERE  name != '/' AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s AND company_id = %(company_id) ORDER BY id DESC LIMIT 1)
odoo13_1  |                 ORDER BY sequence_number DESC
odoo13_1  |                 LIMIT 1
odoo13_1  |             )
odoo13_1  |             RETURNING name;
odoo13_1  |
odoo13_1  | ERROR: unsupported format character '
odoo13_1  | ' (0xa) at index 249
odoo13_1  | 2020-09-29 01:20:10,393 1 ERROR sass13_v2 odoo.http: Exception during JSON request handling.
odoo13_1  | Traceback (most recent call last):
odoo13_1  |   File "/mnt/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
odoo13_1  |     result = request.dispatch()
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 682, in dispatch
odoo13_1  |     result = self._call_function(**self.params)
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 358, in _call_function
odoo13_1  |     return checked_call(self.db, *args, **kwargs)
odoo13_1  |   File "/mnt/odoo/odoo/service/model.py", line 94, in wrapper
odoo13_1  |     return f(dbname, *args, **kwargs)
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 346, in checked_call
odoo13_1  |     result = self.endpoint(*a, **kw)
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 911, in __call__
odoo13_1  |     return self.method(*args, **kw)
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 530, in response_wrap
odoo13_1  |     response = f(*args, **kw)
odoo13_1  |   File "/mnt/odoo/addons/web/controllers/main.py", line 1360, in call_kw
odoo13_1  |     return self._call_kw(model, method, args, kwargs)
odoo13_1  |   File "/mnt/odoo/addons/web/controllers/main.py", line 1352, in _call_kw
odoo13_1  |     return call_kw(request.env[model], method, args, kwargs)
odoo13_1  |   File "/mnt/odoo/odoo/api.py", line 396, in call_kw
odoo13_1  |     result = _call_kw_multi(method, model, args, kwargs)
odoo13_1  |   File "/mnt/odoo/odoo/api.py", line 383, in _call_kw_multi
odoo13_1  |     result = method(recs, *args, **kwargs)
odoo13_1  |   File "/mnt/odoo/addons/account/models/account_move.py", line 1017, in onchange
odoo13_1  |     return super(AccountMove, self.with_context(recursive_onchanges=False)).onchange(values, field_name, field_onchange)
odoo13_1  |   File "/mnt/odoo/odoo/models.py", line 6130, in onchange
odoo13_1  |     for name in nametree
odoo13_1  |   File "/mnt/odoo/odoo/models.py", line 6131, in <listcomp>
odoo13_1  |     if name not in done and snapshot0.has_changed(name)
odoo13_1  |   File "/mnt/odoo/odoo/models.py", line 5985, in has_changed
odoo13_1  |     return self[name] != record[name]
odoo13_1  |   File "/mnt/odoo/odoo/models.py", line 5600, in __getitem__
odoo13_1  |     return self._fields[key].__get__(self, type(self))
odoo13_1  |   File "/mnt/odoo/odoo/fields.py", line 975, in __get__
odoo13_1  |     self.compute_value(recs)
odoo13_1  |   File "/mnt/odoo/odoo/fields.py", line 1114, in compute_value
odoo13_1  |     records._compute_field_value(self)
odoo13_1  |   File "/mnt/odoo/addons/mail/models/mail_thread.py", line 410, in _compute_field_value
odoo13_1  |     return super()._compute_field_value(field)
odoo13_1  |   File "/mnt/odoo/odoo/models.py", line 4005, in _compute_field_value
odoo13_1  |     getattr(self, field.compute)()
odoo13_1  |   File "/mnt/odoo/addons/l10n_latam_invoice_document/models/account_move.py", line 81, in _compute_highest_name
odoo13_1  |     super(AccountMove, self - manual_records)._compute_highest_name()
odoo13_1  |   File "/mnt/odoo/addons/account/models/account_move.py", line 1096, in _compute_highest_name
odoo13_1  |     record.highest_name = record._get_last_sequence()
odoo13_1  |   File "/mnt/odoo/addons/account/models/sequence_mixin.py", line 154, in _get_last_sequence
odoo13_1  |     self.env.cr.execute(query, param)
odoo13_1  |   File "<decorator-gen-3>", line 2, in execute
odoo13_1  |   File "/mnt/odoo/odoo/sql_db.py", line 96, in check
odoo13_1  |     return f(self, *args, **kwargs)
odoo13_1  |   File "/mnt/odoo/odoo/sql_db.py", line 291, in execute
odoo13_1  |     res = self._obj.execute(query, params)
odoo13_1  | Exception
odoo13_1  |
odoo13_1  | The above exception was the direct cause of the following exception:
odoo13_1  |
odoo13_1  | Traceback (most recent call last):
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 638, in _handle_exception
odoo13_1  |     return super(JsonRequest, self)._handle_exception(exception)
odoo13_1  |   File "/mnt/odoo/odoo/http.py", line 314, in _handle_exception
odoo13_1  |     raise exception.with_traceback(None) from new_cause
odoo13_1  | ValueError: unsupported format character '
odoo13_1  | ' (0xa) at index 249
```

I keep investigating tomorrow....but maybe you have an idea what does it happen?